src/ray/gcs/actor_state_accessor.cc

```diff
@@ -0,0 +1,109 @@
+#include "ray/gcs/actor_state_accessor.h"
+#include <boost/none.hpp>
+#include "ray/gcs/redis_gcs_client.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+
+namespace gcs {
+
+ActorStateAccessor::ActorStateAccessor(RedisGcsClient &client_impl)
+    : client_impl_(client_impl) {}
+
+Status ActorStateAccessor::AsyncGet(const JobID &job_id, const ActorID &actor_id,
+                                    const MultiItemCallback<ActorTableData> &callback) {
+  RAY_DCHECK(callback != nullptr);
+  auto on_done = [callback](RedisGcsClient *client, const ActorID &actor_id,
+                            const std::vector<ActorTableData> &data) {
+    callback(Status::OK(), data);
+  };
+
+  ActorTable &actor_table = client_impl_.actor_table();
+  return actor_table.Lookup(job_id, actor_id, on_done);
+}
+
+Status ActorStateAccessor::AsyncAdd(const JobID &job_id, const ActorID &actor_id,
+                                    std::shared_ptr<ActorTableData> data_ptr,
+                                    size_t log_length, const StatusCallback &callback) {
+  ActorTable &actor_table = client_impl_.actor_table();
+  if (callback != nullptr) {
+    auto on_success = [callback, data_ptr](
+        RedisGcsClient *client, const ActorID &actor_id, const ActorTableData &data) {
+      callback(Status::OK());
+    };
+
+    auto on_failure = [callback, data_ptr](
+        RedisGcsClient *client, const ActorID &actor_id, const ActorTableData &data) {
+      callback(Status::Invalid("Add failed, maybe exceeds max reconstruct number."));
+    };
+
+    return actor_table.AppendAt(job_id, actor_id, data_ptr, on_success, on_failure,
```

Comment:
can be simplified a bit by setting `on_success` and `on_failure` accordingly, then `AppendAt` can be called once.

Reply:
There are not many duplicate codes. Both ways are fine.

Reply:
> can be simplified a bit by setting `on_success` and `on_failure` accordingly, then `AppendAt` can be called once.

There are not many duplicate codes. Both ways are fine.